File: src/main/java/app/askresume/oauth/model/OAuthAttributes.java
package app.askresume.oauth.model;

import app.askresume.domain.member.constant.MemberType;
import app.askresume.domain.member.constant.Role;
import app.askresume.domain.member.model.Member;

public record OAuthAttributes(
        String email,
        String name,
        String profile,
        String locale,
        MemberType memberType
) {
    public OAuthAttributes(String email, String name, String profile, String locale, MemberType memberType) {
        this.email = email;
        this.name = name;
        this.profile = profile;
        this.locale = locale;
        this.memberType = memberType;
    }

    public Member toMemberEntity(MemberType memberType, Role role) {
        return Member.builder()
                .username(name)
                .email(email)
                .memberType(memberType)
                .profile(profile)
                .locale(locale)
                .role(role)
                .build();
    }

}

File: src/main/java/app/askresume/oauth/model/OAuthAttributes.kt
package app.askresume.oauth.model

import app.askresume.domain.member.constant.MemberType
import app.askresume.domain.member.constant.Role
import app.askresume.domain.member.model.Member

data class OAuthAttributes(
    val email: String,
    val name: String,
    val profile: String,
    val locale: String,
    val memberType: MemberType,
) {
    fun toMemberEntity(memberType: MemberType, role: Role): Member {
        return Member(
            username = name,
            email = email,
            memberType = memberType,
            profile = profile,
            locale = locale,
            role = role,
        )
    }
}

